core/pkg/worker/instrumented_worker.go

```diff
@@ -38,59 +39,56 @@ func InstrumentedWorkerFn(w Worker) Worker {
 	}
 }
 
-// HandleEvent implements worker.Adapter
-func (a *instrumentedWorker) HandleEvent(ctx context.Context, trigger *triggers.Event) error {
-	var s trace.Span
-
-	ctx, s = otel.Tracer("membrane/pkg/worker", trace.WithInstrumentationVersion(span.MembraneVersion)).
-		Start(ctx, span.Name("topic-"+trigger.Topic))
-
-	s.SetAttributes(
-		semconv.CodeFunctionKey.String("HandleEvent"),
-		semconv.MessagingDestinationKindTopic,
-		semconv.MessagingDestinationKey.String(trigger.Topic),
-		semconv.MessagingMessageIDKey.String(trigger.ID),
-	)
-
-	defer s.End()
-
-	err := a.Worker.HandleEvent(ctx, trigger)
-	if err != nil {
-		s.SetStatus(codes.Error, "Event Handler returned an error")
-		s.RecordError(err)
-	} else {
-		s.SetStatus(codes.Ok, "Event Handled Successfully")
+func (a *instrumentedWorker) tracerFromTrigger(ctx context.Context, trigger *v1.TriggerRequest) (context.Context, trace.Span, error) {
+	if http := trigger.GetHttp(); http != nil {
+		ctx, s := otel.Tracer("membrane/pkg/worker", trace.WithInstrumentationVersion(span.MembraneVersion)).
+			Start(ctx, span.Name(http.Path))
+
+		s.SetAttributes(
+			semconv.CodeFunctionKey.String("HandleHttp"),
+			semconv.HTTPMethodKey.String(http.Method),
+			semconv.HTTPTargetKey.String(http.Path),
+			// semconv.HTTPURLKey.String(http.URL),
+		)
+
+		return ctx, s, nil
+	} else if topic := trigger.GetTopic(); topic != nil {
+		ctx, s := otel.Tracer("membrane/pkg/worker", trace.WithInstrumentationVersion(span.MembraneVersion)).
+			Start(ctx, span.Name("topic-"+topic.Topic))
+
+		s.SetAttributes(
+			semconv.CodeFunctionKey.String("HandleEvent"),
+			semconv.MessagingDestinationKindTopic,
+			semconv.MessagingDestinationKey.String(topic.Topic),
+			// semconv.MessagingMessageIDKey.String(topic.),
```

Comment:
TODO: Replace this

Reply:
Not required so leaving out for now. Have added missing required key for system instead